public/app/plugins/datasource/tempo/metricsSummary.ts

```diff
@@ -0,0 +1,294 @@
+import {
+  DataSourceInstanceSettings,
+  FieldDTO,
+  FieldType,
+  LoadingState,
+  MutableDataFrame,
+  sortDataFrame,
+} from '@grafana/data';
+
+export type MetricsSummary = {
+  spanCount: string;
+  errorSpanCount?: string;
+  p50: string;
+  p90: string;
+  p95: string;
+  p99: string;
+  series: Series[];
+};
+
+type Series = {
+  key: string;
+  value: {
+    type: number;
+    n?: number;
+    f?: number;
+    s?: string;
+    b?: string;
+    d?: string;
+    status?: number;
+    kind?: number;
+  };
+};
+
+type MetricsData = {
+  spanCount: string;
+  errorPercentage: string;
+  p50: string;
+  p90: string;
+  p95: string;
+  p99: string;
+  [key: string]: string | number;
+};
+
+export function createTableFrameFromMetricsSummaryQuery(
+  data: MetricsSummary[],
+  targetQuery: string,
+  instanceSettings: DataSourceInstanceSettings,
+  state: LoadingState
+) {
+  let frame;
+
+  if (state === LoadingState.Error) {
+    frame = emptyResponse;
+  } else if (state === LoadingState.Loading) {
```

Comment:
How do we get to this point? It does not seem like Tempo is streaming the response is it?

Reply:
We get here via the `startWith` that sets the state to `Loading`. The metrics summary endpoint is not queried via steaming.